chassis.go

```diff
@@ -32,66 +33,79 @@ type Chassis struct {
 	VtepLogicalSwitches []string
 }
 
-func (odbi *ovndb) chassisAddImp(name string, hostname string, etype string, ip string) (*OvnCommand, error) {
-	// / Prepare for encap record
-	enCapUUID, err := newRowUUID()
-	if err != nil {
-		return nil, err
+func (odbi *ovndb) chassisAddImp(name string, hostname string, etype []string, ip string,
+	external_ids map[string]string, transport_zones []string, vtep_lswitches []string) (*OvnCommand, error) {
+	if len(name) == 0 {
+		return nil, fmt.Errorf("chassis name cannot be empty")
 	}
-	row := make(OVNRow)
-
-	if len(name) > 0 {
-		row["chassis_name"] = name
+	if len(etype) == 0 {
```

Comment:
What if it is nil? Same for all other validations.

Reply:
for strings nil cannot be assigned but for array,  it works.
package main

import "fmt"

func main (){
	var etype [] string

	if len(etype) == 0 {
	     fmt.Println("len is 0 for empty array")
	}
	etype = nil
	if len(etype) == 0 {
	     fmt.Println("len is 0 for nil")
	}
}
o/p:
len is 0 for empty array
len is 0 for nil